{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "ef792bda",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ba19ec05",
   "metadata": {},
   "source": [
    "### Folders to work\n",
    "\n",
    "We have to use the dataset \"results-with-evidence-full.txt\" from the next folders:\n",
    "- AD (atomic dermatitis) = \"cloud-data/its-cmo-darwin-magellan-workspaces-folders/WS_PMCB/BisCiT_Repository/results/current_version/v2.0c/ad\"\n",
    "- CD (Crohn's Disease) = \"cloud-data/its-cmo-darwin-magellan-workspaces-folders/WS_PMCB/BisCiT_Repository/results/current_version/v2.0c/cd\"\n",
    "- ASA (Asthma) = \"cloud-data/its-cmo-darwin-magellan-workspaces-folders/WS_PMCB/BisCiT_Repository/results/current_version/v2.0c/asa\"\n",
    "- COPD (Chronic Obstructive Pulmonary Disorder) = \"cloud-data/its-cmo-darwin-magellan-workspaces-folders/WS_PMCB/BisCiT_Repository/results/current_version/v2.0c/copd\"\n",
    "- HS (Hidratenitis Suppertiva) = \"cloud-data/its-cmo-darwin-magellan-workspaces-folders/WS_PMCB/BisCiT_Repository/results/current_version/v2.0c/hs\"\n",
    "- PSO (Psoriasis) = \"cloud-data/its-cmo-darwin-magellan-workspaces-folders/WS_PMCB/BisCiT_Repository/results/current_version/v2.0c/pso\"\n",
    "- SLE (Systemic Lupus Erthymatosus) = \"cloud-data/its-cmo-darwin-magellan-workspaces-folders/WS_PMCB/BisCiT_Repository/results/current_version/v2.0c/sle\"\n",
    "- T1D (Type 1 Diabetes) = \"cloud-data/its-cmo-darwin-magellan-workspaces-folders/WS_PMCB/BisCiT_Repository/results/current_version/v2.0c/t1d\"\n",
    "- UC (Ulcerative Colitis) = \"cloud-data/its-cmo-darwin-magellan-workspaces-folders/WS_PMCB/BisCiT_Repository/results/current_version/v2.0c/uc\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "7dcc5307",
   "metadata": {},
   "outputs": [],
   "source": [
    "folders = {\n",
    "    \"AD\": \"cloud-data/its-cmo-darwin-magellan-workspaces-folders/WS_PMCB/BisCiT_Repository/results/current_version/v2.0c/ad/dtl-surfaceome-secretome-equal_0.1\",\n",
    "    \"CD\": \"cloud-data/its-cmo-darwin-magellan-workspaces-folders/WS_PMCB/BisCiT_Repository/results/current_version/v2.0c/cd/dtl-surfaceome-secretome-equal_0.1\",\n",
    "    \"ASA\": \"cloud-data/its-cmo-darwin-magellan-workspaces-folders/WS_PMCB/BisCiT_Repository/results/current_version/v2.0c/asa/dtl-surfaceome-secretome-equal_0.1\",\n",
    "    \"COPD\": \"cloud-data/its-cmo-darwin-magellan-workspaces-folders/WS_PMCB/BisCiT_Repository/results/current_version/v2.0c/copd/dtl-surfaceome-secretome-equal_0.1\",\n",
    "    \"HS\": \"cloud-data/its-cmo-darwin-magellan-workspaces-folders/WS_PMCB/BisCiT_Repository/results/current_version/v2.0c/hs/dtl-surfaceome-secretome-equal_0.1\",\n",
    "    \"PSO\": \"cloud-data/its-cmo-darwin-magellan-workspaces-folders/WS_PMCB/BisCiT_Repository/results/current_version/v2.0c/pso/dtl-surfaceome-secretome-equal_0.1\",\n",
    "    \"SLE\": \"cloud-data/its-cmo-darwin-magellan-workspaces-folders/WS_PMCB/BisCiT_Repository/results/current_version/v2.0c/sle/dtl-surfaceome-secretome-equal_0.1\",\n",
    "    \"T1D\": \"cloud-data/its-cmo-darwin-magellan-workspaces-folders/WS_PMCB/BisCiT_Repository/results/current_version/v2.0c/t1d/dtl-surfaceome-secretome-equal_0.1\",\n",
    "    \"UC\": \"cloud-data/its-cmo-darwin-magellan-workspaces-folders/WS_PMCB/BisCiT_Repository/results/current_version/v2.0c/uc/dtl-surfaceome-secretome-equal_0.1\"\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b9e75b3d",
   "metadata": {},
   "source": [
    "### Training data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "31359577",
   "metadata": {},
   "outputs": [],
   "source": [
    "folder = 'AD'\n",
    "csv_path = '../../../'+ folders[folder] + '/results-with-evidence-full.txt'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "12a3fc98",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'../../../cloud-data/its-cmo-darwin-magellan-workspaces-folders/WS_PMCB/BisCiT_Repository/results/current_version/v2.0c/ad/dtl-surfaceome-secretome-equal_0.1/results-with-evidence-full.txt'"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "csv_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "ccf754e3",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(csv_path, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "f1a509e4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene 1</th>\n",
       "      <th>Gene 2</th>\n",
       "      <th>AD Score</th>\n",
       "      <th>TIE</th>\n",
       "      <th>TIE Bulk RNA</th>\n",
       "      <th>Genetics</th>\n",
       "      <th>Genetics, Rare Variants</th>\n",
       "      <th>Knowledge Graph Connectivity</th>\n",
       "      <th>Pathway Coverage, Positive</th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>...</th>\n",
       "      <th>SC KO, positive</th>\n",
       "      <th>Pathway Coverage, Negative</th>\n",
       "      <th>SC KO, Negative</th>\n",
       "      <th>DTL, Th1-17</th>\n",
       "      <th>DTL, Th2</th>\n",
       "      <th>Immune Competition</th>\n",
       "      <th>CScore</th>\n",
       "      <th>Single Cell Co-dysregulation, positive</th>\n",
       "      <th>Single Cell Co-dysregulation, negative</th>\n",
       "      <th>Single Cell Co-dysregulation, mixed</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IL22</td>\n",
       "      <td>MMP12</td>\n",
       "      <td>5.376584</td>\n",
       "      <td>0.974365</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.593204</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.818245</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.902994</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>72.325136</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.574925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AREG</td>\n",
       "      <td>CCL5</td>\n",
       "      <td>5.126559</td>\n",
       "      <td>0.584504</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.697587</td>\n",
       "      <td>0.572231</td>\n",
       "      <td>0.002588</td>\n",
       "      <td>...</td>\n",
       "      <td>0.273019</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.112872</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.065755</td>\n",
       "      <td>12.633273</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>19.653821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CCL3</td>\n",
       "      <td>MMP9</td>\n",
       "      <td>5.124927</td>\n",
       "      <td>0.662121</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.912395</td>\n",
       "      <td>0.094449</td>\n",
       "      <td>0.000918</td>\n",
       "      <td>...</td>\n",
       "      <td>0.464270</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.056532</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.095987</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>158.680901</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CXCL8</td>\n",
       "      <td>MMP9</td>\n",
       "      <td>5.121890</td>\n",
       "      <td>0.728792</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.972016</td>\n",
       "      <td>0.099010</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.508250</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.065159</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.131484</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>120.994744</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AREG</td>\n",
       "      <td>ITGB2</td>\n",
       "      <td>5.069493</td>\n",
       "      <td>0.515902</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.659660</td>\n",
       "      <td>0.478307</td>\n",
       "      <td>0.164692</td>\n",
       "      <td>...</td>\n",
       "      <td>0.357286</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.175586</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.053436</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>20.436369</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Gene 1 Gene 2  AD Score       TIE  TIE Bulk RNA  Genetics  \\\n",
       "0   IL22  MMP12  5.376584  0.974365           1.0  0.593204   \n",
       "1   AREG   CCL5  5.126559  0.584504           1.0  0.000000   \n",
       "2   CCL3   MMP9  5.124927  0.662121           1.0  0.000000   \n",
       "3  CXCL8   MMP9  5.121890  0.728792           1.0  0.000000   \n",
       "4   AREG  ITGB2  5.069493  0.515902           1.0  0.000000   \n",
       "\n",
       "   Genetics, Rare Variants  Knowledge Graph Connectivity  \\\n",
       "0                      0.0                      0.818245   \n",
       "1                      0.0                      0.697587   \n",
       "2                      0.0                      0.912395   \n",
       "3                      0.0                      0.972016   \n",
       "4                      0.0                      0.659660   \n",
       "\n",
       "   Pathway Coverage, Positive  Proteomics  ...  SC KO, positive  \\\n",
       "0                    0.000000    0.902994  ...         0.000000   \n",
       "1                    0.572231    0.002588  ...         0.273019   \n",
       "2                    0.094449    0.000918  ...         0.464270   \n",
       "3                    0.099010    0.000000  ...         0.508250   \n",
       "4                    0.478307    0.164692  ...         0.357286   \n",
       "\n",
       "   Pathway Coverage, Negative  SC KO, Negative  DTL, Th1-17  DTL, Th2  \\\n",
       "0                         0.0         0.000000          NaN       NaN   \n",
       "1                         0.0        -0.112872          NaN       NaN   \n",
       "2                         0.0        -0.056532          NaN       NaN   \n",
       "3                         0.0        -0.065159          NaN       NaN   \n",
       "4                         0.0        -0.175586          NaN       NaN   \n",
       "\n",
       "   Immune Competition    CScore  Single Cell Co-dysregulation, positive  \\\n",
       "0                 0.0  0.000000                               72.325136   \n",
       "1                 NaN  0.065755                               12.633273   \n",
       "2                 NaN  0.095987                                0.000000   \n",
       "3                 0.0  0.131484                                0.000000   \n",
       "4                 NaN  0.053436                                0.000000   \n",
       "\n",
       "   Single Cell Co-dysregulation, negative  Single Cell Co-dysregulation, mixed  \n",
       "0                                0.000000                             4.574925  \n",
       "1                                0.000000                            19.653821  \n",
       "2                              158.680901                             0.000000  \n",
       "3                              120.994744                             0.000000  \n",
       "4                                0.000000                            20.436369  \n",
       "\n",
       "[5 rows x 26 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "792ec221",
   "metadata": {},
   "source": [
    "### Columns to use\n",
    "\n",
    "The next are the one that are gonna be used to train the models\n",
    "- 'TIE'\n",
    "- 'TIE Bulk RNA'\n",
    "- 'Genetics'\n",
    "- 'Genetics, Rare Variants'\n",
    "- 'Knowledge Graph Connectivity'\n",
    "- 'Pathway Coverage, Positive'\n",
    "- 'Proteomics'\n",
    "- 'Single Cell DE'\n",
    "- 'Single Cell Expression'\n",
    "- 'Single Cell Specificity'\n",
    "- 'Ligand Receptor Significance'\n",
    "- 'Spatial Inflamed vs Noninflamed'\n",
    "- 'Spatial Disease vs Healthy'\n",
    "- 'SC KO, positive'\n",
    "- 'Pathway Coverage, Negative'\n",
    "- 'SC KO, Negative'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "c75dbbee",
   "metadata": {},
   "outputs": [],
   "source": [
    "final_df = df[['Gene 1', 'Gene 2', 'TIE', 'TIE Bulk RNA', 'Genetics',\n",
    "       'Genetics, Rare Variants', 'Knowledge Graph Connectivity',\n",
    "       'Pathway Coverage, Positive', 'Proteomics', 'Single Cell DE',\n",
    "       'Single Cell Expression', 'Single Cell Specificity',\n",
    "       'Ligand Receptor Significance', 'Spatial Inflamed vs Noninflamed',\n",
    "       'Spatial Disease vs Healthy', 'SC KO, positive',\n",
    "       'Pathway Coverage, Negative', 'SC KO, Negative']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "21a1058e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene 1</th>\n",
       "      <th>Gene 2</th>\n",
       "      <th>TIE</th>\n",
       "      <th>TIE Bulk RNA</th>\n",
       "      <th>Genetics</th>\n",
       "      <th>Genetics, Rare Variants</th>\n",
       "      <th>Knowledge Graph Connectivity</th>\n",
       "      <th>Pathway Coverage, Positive</th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Single Cell DE</th>\n",
       "      <th>Single Cell Expression</th>\n",
       "      <th>Single Cell Specificity</th>\n",
       "      <th>Ligand Receptor Significance</th>\n",
       "      <th>Spatial Inflamed vs Noninflamed</th>\n",
       "      <th>Spatial Disease vs Healthy</th>\n",
       "      <th>SC KO, positive</th>\n",
       "      <th>Pathway Coverage, Negative</th>\n",
       "      <th>SC KO, Negative</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IL22</td>\n",
       "      <td>MMP12</td>\n",
       "      <td>0.974365</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.593204</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.818245</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.902994</td>\n",
       "      <td>0.415046</td>\n",
       "      <td>0.195128</td>\n",
       "      <td>0.307783</td>\n",
       "      <td>0.169819</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AREG</td>\n",
       "      <td>CCL5</td>\n",
       "      <td>0.584504</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.697587</td>\n",
       "      <td>0.572231</td>\n",
       "      <td>0.002588</td>\n",
       "      <td>0.175511</td>\n",
       "      <td>0.473162</td>\n",
       "      <td>0.192210</td>\n",
       "      <td>0.542481</td>\n",
       "      <td>0.609430</td>\n",
       "      <td>0.116707</td>\n",
       "      <td>0.273019</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.112872</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CCL3</td>\n",
       "      <td>MMP9</td>\n",
       "      <td>0.662121</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.912395</td>\n",
       "      <td>0.094449</td>\n",
       "      <td>0.000918</td>\n",
       "      <td>0.432339</td>\n",
       "      <td>0.374286</td>\n",
       "      <td>0.505097</td>\n",
       "      <td>0.522113</td>\n",
       "      <td>0.213472</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.464270</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.056532</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CXCL8</td>\n",
       "      <td>MMP9</td>\n",
       "      <td>0.728792</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.972016</td>\n",
       "      <td>0.099010</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.499769</td>\n",
       "      <td>0.446458</td>\n",
       "      <td>0.563021</td>\n",
       "      <td>0.336794</td>\n",
       "      <td>0.032941</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.508250</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.065159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AREG</td>\n",
       "      <td>ITGB2</td>\n",
       "      <td>0.515902</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.659660</td>\n",
       "      <td>0.478307</td>\n",
       "      <td>0.164692</td>\n",
       "      <td>0.192836</td>\n",
       "      <td>0.579722</td>\n",
       "      <td>0.082402</td>\n",
       "      <td>0.637007</td>\n",
       "      <td>0.463761</td>\n",
       "      <td>0.113504</td>\n",
       "      <td>0.357286</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.175586</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Gene 1 Gene 2       TIE  TIE Bulk RNA  Genetics  Genetics, Rare Variants  \\\n",
       "0   IL22  MMP12  0.974365           1.0  0.593204                      0.0   \n",
       "1   AREG   CCL5  0.584504           1.0  0.000000                      0.0   \n",
       "2   CCL3   MMP9  0.662121           1.0  0.000000                      0.0   \n",
       "3  CXCL8   MMP9  0.728792           1.0  0.000000                      0.0   \n",
       "4   AREG  ITGB2  0.515902           1.0  0.000000                      0.0   \n",
       "\n",
       "   Knowledge Graph Connectivity  Pathway Coverage, Positive  Proteomics  \\\n",
       "0                      0.818245                    0.000000    0.902994   \n",
       "1                      0.697587                    0.572231    0.002588   \n",
       "2                      0.912395                    0.094449    0.000918   \n",
       "3                      0.972016                    0.099010    0.000000   \n",
       "4                      0.659660                    0.478307    0.164692   \n",
       "\n",
       "   Single Cell DE  Single Cell Expression  Single Cell Specificity  \\\n",
       "0        0.415046                0.195128                 0.307783   \n",
       "1        0.175511                0.473162                 0.192210   \n",
       "2        0.432339                0.374286                 0.505097   \n",
       "3        0.499769                0.446458                 0.563021   \n",
       "4        0.192836                0.579722                 0.082402   \n",
       "\n",
       "   Ligand Receptor Significance  Spatial Inflamed vs Noninflamed  \\\n",
       "0                      0.169819                         0.000000   \n",
       "1                      0.542481                         0.609430   \n",
       "2                      0.522113                         0.213472   \n",
       "3                      0.336794                         0.032941   \n",
       "4                      0.637007                         0.463761   \n",
       "\n",
       "   Spatial Disease vs Healthy  SC KO, positive  Pathway Coverage, Negative  \\\n",
       "0                    0.000000         0.000000                         0.0   \n",
       "1                    0.116707         0.273019                         0.0   \n",
       "2                    0.000000         0.464270                         0.0   \n",
       "3                    0.000000         0.508250                         0.0   \n",
       "4                    0.113504         0.357286                         0.0   \n",
       "\n",
       "   SC KO, Negative  \n",
       "0         0.000000  \n",
       "1        -0.112872  \n",
       "2        -0.056532  \n",
       "3        -0.065159  \n",
       "4        -0.175586  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d017c706",
   "metadata": {},
   "source": [
    "### Target data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "fd85f8db",
   "metadata": {},
   "outputs": [],
   "source": [
    "competition_score_path = '../../dataset/GENS_DEVELOPMENT_SCORE.csv'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2da4cb90",
   "metadata": {},
   "outputs": [],
   "source": [
    "#comp_df = pd.read_csv(competition_score_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "662c1ed9",
   "metadata": {},
   "outputs": [],
   "source": [
    "#comp_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3ac08429",
   "metadata": {},
   "source": [
    "Calculate the croos product of every gen"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "c99ca62d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total de genes: 58786\n",
      "Total de pares únicos: 1,727,926,291\n",
      "Procesado: 0 / 58,786 genes\n",
      "Procesado: 1,000 / 58,786 genes\n",
      "Procesado: 2,000 / 58,786 genes\n",
      "Procesado: 3,000 / 58,786 genes\n",
      "Procesado: 4,000 / 58,786 genes\n",
      "Procesado: 5,000 / 58,786 genes\n",
      "Procesado: 6,000 / 58,786 genes\n",
      "Procesado: 7,000 / 58,786 genes\n",
      "Procesado: 8,000 / 58,786 genes\n",
      "Procesado: 9,000 / 58,786 genes\n",
      "Procesado: 10,000 / 58,786 genes\n",
      "Procesado: 11,000 / 58,786 genes\n",
      "Procesado: 12,000 / 58,786 genes\n",
      "Procesado: 13,000 / 58,786 genes\n",
      "Procesado: 14,000 / 58,786 genes\n",
      "Procesado: 15,000 / 58,786 genes\n",
      "Procesado: 16,000 / 58,786 genes\n",
      "Procesado: 17,000 / 58,786 genes\n",
      "Procesado: 18,000 / 58,786 genes\n",
      "Procesado: 19,000 / 58,786 genes\n",
      "Procesado: 20,000 / 58,786 genes\n",
      "Procesado: 21,000 / 58,786 genes\n",
      "Procesado: 22,000 / 58,786 genes\n",
      "Procesado: 23,000 / 58,786 genes\n",
      "Procesado: 24,000 / 58,786 genes\n",
      "Procesado: 25,000 / 58,786 genes\n",
      "Procesado: 26,000 / 58,786 genes\n",
      "Procesado: 27,000 / 58,786 genes\n",
      "Procesado: 28,000 / 58,786 genes\n",
      "Procesado: 29,000 / 58,786 genes\n",
      "Procesado: 30,000 / 58,786 genes\n",
      "Procesado: 31,000 / 58,786 genes\n",
      "Procesado: 32,000 / 58,786 genes\n",
      "Procesado: 33,000 / 58,786 genes\n",
      "Procesado: 34,000 / 58,786 genes\n",
      "Procesado: 35,000 / 58,786 genes\n",
      "Procesado: 36,000 / 58,786 genes\n",
      "Procesado: 37,000 / 58,786 genes\n",
      "Procesado: 38,000 / 58,786 genes\n",
      "Procesado: 39,000 / 58,786 genes\n",
      "Procesado: 40,000 / 58,786 genes\n",
      "Procesado: 41,000 / 58,786 genes\n",
      "Procesado: 42,000 / 58,786 genes\n",
      "Procesado: 43,000 / 58,786 genes\n",
      "Procesado: 44,000 / 58,786 genes\n",
      "Procesado: 45,000 / 58,786 genes\n",
      "Procesado: 46,000 / 58,786 genes\n",
      "Procesado: 47,000 / 58,786 genes\n",
      "Procesado: 48,000 / 58,786 genes\n",
      "Procesado: 49,000 / 58,786 genes\n",
      "Procesado: 50,000 / 58,786 genes\n",
      "Procesado: 51,000 / 58,786 genes\n",
      "Procesado: 52,000 / 58,786 genes\n",
      "Procesado: 53,000 / 58,786 genes\n",
      "Procesado: 54,000 / 58,786 genes\n",
      "Procesado: 55,000 / 58,786 genes\n",
      "Procesado: 56,000 / 58,786 genes\n",
      "Procesado: 57,000 / 58,786 genes\n",
      "Procesado: 58,000 / 58,786 genes\n",
      "Completado! Archivo guardado en: unique_pairs.csv\n"
     ]
    }
   ],
   "source": [
    "def create_unique_pairs_batched(df, batch_size=1000, output_file='unique_pairs.csv'):\n",
    "    genes_data = df[['gene_symbol', 'development_score_log_norm']].copy()\n",
    "    n_genes = len(genes_data)\n",
    "    \n",
    "    print(f\"Total de genes: {n_genes}\")\n",
    "    print(f\"Total de pares únicos: {n_genes * (n_genes + 1) // 2:,}\")\n",
    "    \n",
    "    first_batch = True\n",
    "    \n",
    "    for i in range(n_genes):\n",
    "        # Solo combinar con genes desde i en adelante\n",
    "        genes_from_i = genes_data.iloc[i:].copy()\n",
    "        gene_i = genes_data.iloc[i]\n",
    "        \n",
    "        result_batch = pd.DataFrame({\n",
    "            'gene_symbol_1': gene_i['gene_symbol'],\n",
    "            'gene_symbol_2': genes_from_i['gene_symbol'].values,\n",
    "            'score_product': gene_i['development_score_log_norm'] * genes_from_i['development_score_log_norm'].values\n",
    "        })\n",
    "        \n",
    "        result_batch.to_csv(\n",
    "            output_file,\n",
    "            mode='a' if not first_batch else 'w',\n",
    "            header=first_batch,\n",
    "            index=False\n",
    "        )\n",
    "        \n",
    "        first_batch = False\n",
    "        \n",
    "        if i % 1000 == 0:\n",
    "            print(f\"Procesado: {i:,} / {n_genes:,} genes\")\n",
    "    \n",
    "    print(f\"Completado! Archivo guardado en: {output_file}\")\n",
    "\n",
    "# Usar la función\n",
    "create_unique_pairs_batched(comp_df, output_file='unique_pairs.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "eaf7dc89",
   "metadata": {},
   "outputs": [],
   "source": [
    "combined_genes_score_path = \"unique_pairs.csv\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "038b1aa0",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/local/anaconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3326: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    }
   ],
   "source": [
    "combined_df = pd.read_csv(combined_genes_score_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "4a7cb07d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_symbol_1</th>\n",
       "      <th>gene_symbol_2</th>\n",
       "      <th>score_product</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IL22</td>\n",
       "      <td>IL22</td>\n",
       "      <td>0.000100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IL22</td>\n",
       "      <td>MMP12</td>\n",
       "      <td>0.000793</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IL22</td>\n",
       "      <td>IL18RAP</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IL22</td>\n",
       "      <td>CSF2</td>\n",
       "      <td>0.009871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IL22</td>\n",
       "      <td>CCL17</td>\n",
       "      <td>0.002256</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  gene_symbol_1 gene_symbol_2  score_product\n",
       "0          IL22          IL22       0.000100\n",
       "1          IL22         MMP12       0.000793\n",
       "2          IL22       IL18RAP       0.000000\n",
       "3          IL22          CSF2       0.009871\n",
       "4          IL22         CCL17       0.002256"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "84ec9347",
   "metadata": {},
   "source": [
    "### Synergy data\n",
    "\n",
    "It can be added to the training data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "bd967f37",
   "metadata": {},
   "outputs": [],
   "source": [
    "synergy_path = \"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "325cb9b3",
   "metadata": {},
   "outputs": [],
   "source": [
    "if folder in [\"HS\", \"T1D\", \"CD\", \"UC\", \"SLE\", \"PSO\"]:\n",
    "    synergy_path = \"../../../cloud-data/its-cmo-darwin-magellan-workspaces-folders/WS_PMCB/BisCiT_Repository/scoring/M6_Validation/DTL/na.dtl.pbmc.pbmc.dtl_first.cytokine.custom.th1-17.hgnc.either.2d.txt.gz\"\n",
    "else: \n",
    "    synergy_path = \"../../../cloud-data/its-cmo-darwin-magellan-workspaces-folders/WS_PMCB/BisCiT_Repository/scoring/M6_Validation/DTL/na.dtl.pbmc.pbmc.dtl_first.cytokine.custom.th2.hgnc.either.2d.txt.gz\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "4762c778",
   "metadata": {},
   "outputs": [],
   "source": [
    "synergy_df = pd.read_csv(synergy_path, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "ee600069",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hgnc_id</th>\n",
       "      <th>AOC3</th>\n",
       "      <th>APOA1</th>\n",
       "      <th>CD19</th>\n",
       "      <th>CD2</th>\n",
       "      <th>CD28</th>\n",
       "      <th>CD38</th>\n",
       "      <th>CD40LG</th>\n",
       "      <th>CD52</th>\n",
       "      <th>CD6</th>\n",
       "      <th>...</th>\n",
       "      <th>TLR4</th>\n",
       "      <th>TNF</th>\n",
       "      <th>TNFRSF4</th>\n",
       "      <th>TNFSF12</th>\n",
       "      <th>TNFSF13</th>\n",
       "      <th>TNFSF13B</th>\n",
       "      <th>TNFSF14</th>\n",
       "      <th>TNFSF15</th>\n",
       "      <th>TNFSF4</th>\n",
       "      <th>TSLP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AOC3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.149503</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.433832</td>\n",
       "      <td>-0.497219</td>\n",
       "      <td>-0.946818</td>\n",
       "      <td>-0.217865</td>\n",
       "      <td>...</td>\n",
       "      <td>0.484279</td>\n",
       "      <td>-0.044846</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.252261</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.304771</td>\n",
       "      <td>-0.018014</td>\n",
       "      <td>0.030056</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>APOA1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.050800</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.106959</td>\n",
       "      <td>-0.065580</td>\n",
       "      <td>-0.139641</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.216877</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.244687</td>\n",
       "      <td>0.192409</td>\n",
       "      <td>-0.158345</td>\n",
       "      <td>-0.623019</td>\n",
       "      <td>0.462384</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CD19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.441647</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>-0.085939</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.197634</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.027051</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CD2</td>\n",
       "      <td>0.391909</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.440277</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.184906</td>\n",
       "      <td>-0.318333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.195465</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.227356</td>\n",
       "      <td>0.399483</td>\n",
       "      <td>0.430062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.340099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CD28</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.701815</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.364797</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.084447</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.167012</td>\n",
       "      <td>-0.573543</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.404279</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 65 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  hgnc_id      AOC3  APOA1      CD19       CD2  CD28      CD38    CD40LG  \\\n",
       "0    AOC3       NaN    NaN       NaN  0.149503   NaN -0.433832 -0.497219   \n",
       "1   APOA1       NaN    NaN       NaN -0.050800   NaN -0.106959 -0.065580   \n",
       "2    CD19       NaN    NaN       NaN  0.441647   NaN  1.000000 -0.085939   \n",
       "3     CD2  0.391909    NaN  0.440277       NaN   NaN       NaN  0.184906   \n",
       "4    CD28       NaN    NaN  0.701815       NaN   NaN  0.364797       NaN   \n",
       "\n",
       "       CD52       CD6  ...      TLR4       TNF  TNFRSF4  TNFSF12  TNFSF13  \\\n",
       "0 -0.946818 -0.217865  ...  0.484279 -0.044846      NaN      NaN      NaN   \n",
       "1 -0.139641       NaN  ... -0.216877       NaN      NaN      NaN      NaN   \n",
       "2       NaN  0.197634  ...       NaN -0.027051      NaN      NaN      NaN   \n",
       "3 -0.318333       NaN  ...       NaN  0.195465      NaN      NaN      NaN   \n",
       "4       NaN       NaN  ...       NaN  0.084447      NaN      NaN      NaN   \n",
       "\n",
       "   TNFSF13B   TNFSF14   TNFSF15    TNFSF4      TSLP  \n",
       "0  0.252261       NaN  0.304771 -0.018014  0.030056  \n",
       "1  0.244687  0.192409 -0.158345 -0.623019  0.462384  \n",
       "2       NaN       NaN       NaN       NaN       NaN  \n",
       "3  0.227356  0.399483  0.430062       NaN  0.340099  \n",
       "4       NaN  0.167012 -0.573543       NaN  0.404279  \n",
       "\n",
       "[5 rows x 65 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "synergy_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6ffe21c2",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
